packages/workbox-google-analytics/src/index.js

```diff
@@ -56,83 +49,122 @@ import replayQueuedRequests from './lib/replay-queued-requests.js';
+        new URL(request.url).searchParams;
+
+    // Set the qt param prior to apply the hitFilter or parameterOverrides.
+    params.set('qt', queueTime);
+
+    if (config.parameterOverrides) {
+      for (const param of Object.keys(config.parameterOverrides)) {
+        const value = config.parameterOverrides[param];
+        params.set(param, value);
       }
     }
+
+    if (typeof config.hitFilter == 'function') {
+      config.hitFilter.call(null, params);
+    }
+
+    request.body = params.toString();
+    request.method = 'POST';
+    request.mode = 'cors';
+    request.credentials = 'omit';
+    request.headers = '[["Content-Type", "text/plain"]]';
+    request.url = `https://${constants.URL.HOST}${constants.URL.COLLECT_PATH}`;
+  };
+};
+
+/**
+ * Creates GET and POST routes to catch failed Measurement Protocol hits.
+ * @param {Function} requestWillDequeueCallback The callback used by
+ *     background sync when requests are dequeued and retried.
+ * @return {Array<Route>} The created routes.
+ */
+const createCollectRoutes = (requestWillDequeueCallback) => {
+  const bgSyncQueuePlugin = new QueuePlugin({
+    dbName: constants.IDB.NAME,
+    maxRetentionTime: constants.STOP_RETRYING_AFTER,
+    callbacks: {requestWillDequeue: requestWillDequeueCallback},
   });
 
-  replayQueuedRequests(config);
+  const match = ({url}) => url.hostname == constants.URL.HOST &&
+      url.pathname == constants.URL.COLLECT_PATH;
+
+  const handler = new NetworkOnly({
+    requestWrapper: new RequestWrapper({plugins: [bgSyncQueuePlugin]}),
+  });
+
+  return [
+    new Route({method: 'GET', match, handler}),
+    new Route({method: 'POST', match, handler}),
+  ];
+};
+
+/**
+ * Creates a route with a network first strategy for the analytics.js script.
+ * @return {Route} The created route.
+ */
+const createAnalyticsJsRoute = () => {
+  return new Route({
+    match: ({url}) => {
+      return url.hostname == constants.URL.HOST &&
+          url.pathname == constants.URL.ANALYTICS_JS_PATH;
+    },
+    handler: new NetworkFirst({
+      requestWrapper: new RequestWrapper({cacheName: constants.CACHE_NAME}),
+    }),
+  });
+};
+
+/**
+ * @alias workbox.googleAnalytics.initialize
+ * @param {Object=} [config]
+ * @param {Object} [config.parameterOverrides]
+ *     [Measurement Protocol parameters](https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters),
+ *     expressed as key/value pairs, to be added to replayed Google Analytics
+ *     requests. This can be used to, e.g., set a custom dimension indicating
+ *     that the request was replayed.
+ * @param {Function} [config.hitFilter] A function that allows you to modify
+ *     the hit parameters prior to replaying
+ *     the hit. The function is invoked with the original hit's URLSearchParams
+ *     object as its only argument.
+ * @memberof module:workbox-google-analytics
+ */
+export const initialize = (config = {}) => {
```

Comment:
what happens if I call this multiple times?

Reply:
Why shouldn't we protect them from it?

Reply:
Only the first `fetch` event handler would end up being invoked, since it would call `respondWith()` when an incoming request matches a `Route`. So there's no reason why you'd want to call `initialize()` multiple times, but there wouldn't be any effective negative impact.